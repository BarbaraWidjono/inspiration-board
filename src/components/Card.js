import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

const Card = (props) => {
  const { id, text, emoji } = props.card;


  return (
    <div className="card">
      <p className="card__content">{text}</p>
      <p>{emoji}</p>
    </div>
  )
}

Card.propTypes = {

};

export default Card;

// class Card extends Component {
//
//   constructor(props){
//     super(props)
//
//     this.state = {
//       allCards: this.props.allCards,
//     }
//   }
//
//
//
//   render() {
//
//     console.log(this.state.allCards);
//
//     return (
//       <div className="card">
//         Card
//       </div>
//     )
//   }
// }
//
// Card.propTypes = {
//
// };
//
// export default Card;
