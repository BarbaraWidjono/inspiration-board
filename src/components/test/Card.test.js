import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('matches existing snapshot', () => {
    const testCard = {
      id: 1,
      text: "yummy",
      emoji: "pizza",
    }

    const testCallback = () => {}

    const wrapper = shallow(
      <Card
      card={testCard}
      removeCardCallback={testCallback} />
      )

    expect(wrapper).toMatchSnapshot();
  })

})
