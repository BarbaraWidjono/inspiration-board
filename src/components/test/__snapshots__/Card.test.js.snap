// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches existing snapshot 1`] = `
<div
  className="card"
>
  <p
    className="card__content"
  >
    yummy
  </p>
  <Emoji
    chars="pizza"
  />
  <button
    className="card__delete"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
</div>
`;
